mitodl/mit-open
# ETL pipeline for MIT edX programs

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5389

### Description (What does it do?)
Adds an ETL pipeline for ingesting MIT edX programs from an API endpoint

### Screenshots (if appropriate):


![Screenshot 2024-09-09 150424](https://github.com/user-attachments/assets/edb958b1-dd94-473d-9583-80e9a48af55f)

### How can this be tested?
- Set the following settings in your backend env (same values as RC if not specified here):
  ```
  EDX_API_ACCESS_TOKEN_URL=https://api.edx.org/oauth2/v1/access_token
  EDX_API_CLIENT_ID=
  EDX_API_CLIENT_SECRET=
  EDX_API_URL=https://api.edx.org/catalog/v1/catalogs/10/courses
  EDX_PROGRAMS_API_URL=https://discovery.edx.org/api/v1/programs/
  ```
- Run `./manage.py backpopulate_edx_data`
- Go to http://open.odl.local:8062/search/?offered_by=mitx&resource_category=program&platform=edx&sortby=new
- The 3 new programs at the top should be "Future Energy Systems", "Computational Thinking using Python", and "Circuits and Electronics"
- The price shown for each program should equal the amount shown on the program url page, before discount.  No idea where the discount comes from, it doesn't seem to be present in the API output.
- Each should have topics and instructors based on the child courses.


### Additional Context
The programs API endpoint also returns micromasters programs, which are currently ignored in favor of the data from the micromasters API endpoint.
